README.md

```diff
@@ -1,70 +1,72 @@
-may be seen in [stub files](./stubs/weak).
+There are no new INI settings, constants, nor exceptions. 
 
-Short list if what provided by this extension is:
+These new classes/functions are implemented through C code, but [PHP stubs with annotations are available](./stubs/weak) for use with IDEs and code-analysis tools. (Also see "Stub files for Composer".)
 
-  - `class Weak\Reference`
-  - `function Weak\refcounted()`
-  - `function Weak\refcount()`
-  - `function Weak\weakrefcounted()`
-  - `function Weak\weakrefcount()`
-  - `function Weak\weakrefs()`
-  - `function Weak\object_handle()`
+### Known issues and limitations
 
-### Special cases
+#### Callbacks will not be triggered if there are uncaught exceptions in the destruction process
 
-#### Notify callback will not be called if any uncatch exception thrown during object destructing process before it.
+A notification callback may fail to be triggered if:
+* The target-object had an unhandled-exception during destruction
+* Any other notification callback for the same object threw an unhandled exception
 
-If any exception in referent object destructor or other weak references notifier callback (if any) during
-object destructing process thrown, next notify callback will not be called.
+Internally, `Weak\Reference`'s notification-callback mechanism may be considered as an extension to the target-object's destruction process.
  
-Internally, `Weak\Reference`'s notify callback may be considered as a destructor extension, something like extending object
-destructor and call notify callbacks one by one in a row, if any. 
+If you re affected by this limitation and need to catch an exception from object destruction, please [open a github issue](https://github.com/pinepain/php-weak/issues/new) to discuss how it may be resolved.
 
-Anyway, we assume that exception in destructors are fatal and the only way they should be handled is fail-fast approach.
- 
-If you fall down under this limitation and have to try-catch object destructing, pleas, fill an issue and we'll discuss
-how this issue may be solved. One of pre-release features was storing destructed object's weak reference in some array
-because storing into array is completely internals process and can not be normally interrupted during standard destructor
-execution (we don't count `die()` before calling notify callback at all.
+#### The target-object's `spl_object_hash()` hash will change each time a `Weak\Reference` created with it
 
-#### Object's `spl_object_hash()` hash changes after `Weak\Reference` created for it.
+Creating a `Weak\Reference` has a side-effect on on the object that you are making the reference to. Because it changes certain internal PHP pointers, it will alter the `spl_object_hash()` of the object. 
 
-Internally `Weak\Reference` changes object handlers pointer, so `spl_object_hash()` function will provide different hash
```

Comment:
Agree that mention such details may be out of scope, but weak references feature targets pretty experienced developers. While it is probably out of readme scope at this time I prefer to keep it.